packages/gatsby/src/schema/infer/__tests__/infer.js

```diff
@@ -1385,4 +1386,44 @@ Object {
       )
     })
   })
+
+  describe(`missing extension warning`, () => {
+    it(`warns when inferred extension is missing in type definition`, async () => {
+      const nodes = [
+        {
+          date: `2012-11-01`,
+          internal: { type: `Test` },
+          id: `1`,
+        },
+        {
+          linked___NODE: `foo`,
+          internal: { type: `Test` },
+          id: `2`,
+        },
+
+        // linked node:
+        { id: `foo`, internal: { type: `Foo` } },
+      ]
+      const typeDefs = [
+        {
+          typeOrTypeDef: `
+            type Test implements Node {
+              linked: Foo
+              date: Date
+            }
+          `,
+        },
+      ]
+      await buildTestSchema(nodes, {}, typeDefs)
+      expect(report.warn.mock.calls.length).toEqual(2)
+      expect(report.warn.mock.calls[0][0]).toEqual(
+        `Deprecation warning - adding inferred extension {"dateformat":{}} for field Test.date. ` +
+          `In Gatsby v3, only fields with an explicit directive/extension will get a resolver.`
+      )
+      expect(report.warn.mock.calls[1][0]).toEqual(
+        `Deprecation warning - adding inferred extension {"link":{"by":"id","from":"linked___NODE"}} for field Test.linked. ` +
```

Comment:
Is it feasible to expand the warning to explain how to add those?

This from my understanding could be copy&pasted into field config when using `schema.buildObjectType` programmatically, but using SDL it would be `@link(by: "id", from: "linked___NODE")`, correct?

Reply:
@pieh Added SDL instructions for consistency with another similar PR: #28532